rest-ehr-scape/src/main/java/org/ehrbase/rest/ehrscape/controller/CompositionController.java

```diff
@@ -62,11 +65,17 @@ public CompositionController(CompositionService compositionService) {
             throw new InvalidApiParameterException(String.format("Template Id needs to specified for format %s", format));
         }
 
-        UUID compositionId = compositionService.create(ehrId, content, format, templateId, linkId);
+        var compoObj = compositionService.buildComposition(content, format, templateId);
+
+        Optional<CompositionDto> optionalCompositionDto = compositionService.create(ehrId, compoObj);
+
+        var compositionUuid = optionalCompositionDto.orElseThrow(() ->
+            new InternalServerException("Failed to create composition"))
+            .getUuid();
 
         CompositionWriteRestResponseData responseData = new CompositionWriteRestResponseData();
         responseData.setAction(Action.CREATE);
-        responseData.setCompositionUid(compositionId + "::" + 1);
+        responseData.setCompositionUid(compositionUuid + "::" + 1);
```

Comment:
@jakesmolka missing the creating_system_id in the OBJECT_VERSION_ID?

https://specifications.openehr.org/releases/UML/latest/index.html#Architecture___18_5_1_83e026d_1503073094521_559965_7682

Reply:
Well, yes but no :smile: This is in the scope of the EHRscape API. And on the one hand, this is out of scope as I was just changing the DTOs due to refactoring here, and on the other hand, this particular fix is waiting over there already: https://github.com/ehrbase/ehrbase/pull/520
(FYI: Renaud just filed the fix as PR to Archie yesterday, so #520 should be able to move forward soon.)

But good spotting there, thanks!